### DISCUSSION TITLE
Correct wiring diagram for programming

### DISCUSSION THREAD

**Discussion by janczmok1970** [PricelessToolkit/ESPBell-MAX#3] [Q&A] [answered]
I tried programming the ESPBell, but yet i fail at successfully doing that.
I connected 3,3V,GND,TX,RX and on the serial monitor i receive:

13:56:28.626 -> failed, rc=-2 try again in 0.05 seconds
13:56:28.665 -> Attempting MQTT connection...Buffer Size increased to 380 byte

but programming is not successfull, i get

. Variables and constants in RAM (global, static), used 29792 / 80192 bytes (37%)
║   SEGMENT  BYTES    DESCRIPTION
╠══ DATA     1512     initialized variables
╠══ RODATA   2336     constants       
╚══ BSS      25944    zeroed variables
. Instruction RAM (IRAM_ATTR, ICACHE_RAM_ATTR), used 60331 / 65536 bytes (92%)
║   SEGMENT  BYTES    DESCRIPTION
╠══ ICACHE   32768    reserved space for flash instruction cache
╚══ IRAM     27563    code in IRAM    
. Code in flash (default, ICACHE_FLASH_ATTR), used 251112 / 1048576 bytes (23%)
║   SEGMENT  BYTES    DESCRIPTION
╚══ IROM     251112   code in flash   
esptool.py v3.0
Serial port /dev/cu.usbserial-0001
Connecting........_____....._____....._____....._____....._____....._____.....____
A fatal esptool.py error occurred: Failed to connect to ESP8266: Timed out waiting for packet header_

Serial TTL works fine with other hardware.

Please advise

**Comment by janczmok1970** [comment, marked as answer]
Answer to myself: Need to press flash button before powering the device.

**Reply by PricelessToolkit** [reply]
Yes :)
